destination/writer/writer.go

```diff
@@ -0,0 +1,163 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package writer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+const (
+	// idFieldName contains default reserved primary key from MongoDB.
+	idFieldName = "_id"
+
+	// setCommand contains command, that used during Update query.
+	setCommand = "$set"
+)
+
+// Writer implements a writer logic for Mongo destination.
+type Writer struct {
+	collection *mongo.Collection
+}
+
+// NewWriter creates new instance of the Writer.
+func NewWriter(ctx context.Context, collection *mongo.Collection) *Writer {
```

Comment:
Do we actually need the `context.Context` here?

Why not just:
```go
// NewWriter creates new instance of the [Writer].
func NewWriter(collection *mongo.Collection) *Writer {
	return &Writer{collection}
}
```

Reply:
fixed